File: 012 Seection Sort/Selection Sort.js
// O(n^2) Time and O(1) Space best, average and worst.

function selectionSort(array) {
	let currentIndex = 0;
	while (currentIndex < array.length - 1) {
		let smallestIndex = currentIndex;
		for (let i = currentIndex + 1; i < array.length; i++ ){
			if (array[smallestIndex] > array[i]) smallestIndex = i;
		}
			swap(currentIndex, smallestIndex, array);
			currentIndex++;
	}
	return array;
}

function swap(a,b,arr){
	[arr[a],arr[b]] = [arr[b],arr[a]]
}

File: 012 Seection Sort/Selection Sort.py
# O(n^2) Time and O(1) Space best, average and worst.

def swap(x,y,arr):
	arr[x], arr[y] = arr[y], arr[x]

def selectionSort(array):
	currIdx = 0 
	while currIdx < len(array)-1:
		smallestIdx = currIdx
		for x in range(currIdx+1, len(array)):
			if arr[smallesIdx] > arr[x]:
				smallestIdx = x
		swap(currIdx, smallestIdx,arr)
		currIdx += 1
	return array